### DISCUSSION TITLE
How do I run the run.py file with the extra training arguments?

### DISCUSSION THREAD

**Discussion by Ednaordinary** [NVlabs/instant-ngp#1375] [Q&A] [answered]
Is there a way to run the script to train with the rand-level, envmap, extrinsics, distortion, and per-image latents like you can within the executable gui? There are no immediate arguments in the arg parser
![image](https://github.com/NVlabs/instant-ngp/assets/88869424/99e0d801-00ba-4fd0-b1c3-2a14791ab74c)

**Comment by Ednaordinary** [comment, marked as answer]
Python bindings at https://github.com/NVlabs/instant-ngp/blob/master/src/python_api.cu
optimize extrinsics (definition of line 651 when writing) would be testbed.nerf.training.optimize_extrinsics = True

**Comment by Timothy-pop** [comment]
"Is it possible to set a 3D rectangular bounding box for the area of interest from the command line, as well as to set the corresponding parameters to export the Mesh?"
